File: bench/gc/src/rebuild.go
// GC 敏感形状 2：每轮重建（Go）。
package main

import "fmt"

type node struct {
v    int64
next *node
}

func main() {
const rounds, k = 20000, 1000
var acc int64
for r := 0; r < rounds; r++ {
var h *node
for i := int64(0); i < k; i++ {
h = &node{v: i, next: h}     // 上一轮的链在这一轮不再被引用 ⇒ 可回收
}
for p := h; p != nil; p = p.next {
acc += p.v
}
}
fmt.Println(acc)
}

File: bench/gc/src/rebuild.rs
// GC 敏感形状 2：每轮重建（Rust，Box 链，每轮自动 drop ⇒ 递归释放）。
struct Node { v: i64, next: Option<Box<Node>> }

fn main() {
    const ROUNDS: i64 = 20000;
    const K: i64 = 1000;
    let mut acc: i64 = 0;
    for _ in 0..ROUNDS {
        let mut h: Option<Box<Node>> = None;
        for i in 0..K {
            h = Some(Box::new(Node { v: i, next: h }));
        }
        let mut p = &h;
        while let Some(n) = p {
            acc += n.v;
            p = &n.next;
        }
    }
    println!("{}", acc);
}
